platform/ios/src/MLNMapView.mm

```diff
@@ -6811,6 +6812,119 @@ - (void)updateHeadingForDeviceOrientation
     return MLNFeaturesFromMBGLFeatures(features);
 }
 
+// MARK: Managing Feature State
+
+- (void)setFeatureState:(id<MLNFeature>)feature sourceID:(NSString *)sourceID state:(NSDictionary<NSString *, id> *)state {
+    MLNAssertIsMainThread();
+
+    if (!feature || !sourceID || !state) {
+        return;
+    }
+
+    NSString *featureID = [NSString stringWithFormat:@"%@", feature.identifier];
+    [self setFeatureStateForSource:sourceID sourceLayer:nil featureID:featureID state:state];
+}
+
+- (void)setFeatureStateForSource:(NSString *)sourceID
+                    sourceLayer:(nullable NSString *)sourceLayerID
+                      featureID:(NSString *)featureID
+                           state:(NSDictionary<NSString *, id> *)state {
+    MLNAssertIsMainThread();
+
+    if (!sourceID || !featureID || !state) {
+        return;
+    }
+
+    mbgl::FeatureState mbglState;
+    for (NSString *key in state) {
+        id value = state[key];
+        NSExpression *expression = [NSExpression expressionForConstantValue:value];
+        mbglState[key.UTF8String] = expression.mgl_constantMBGLValue;
+    }
+
+    _rendererFrontend->getRenderer()->setFeatureState(
+        sourceID.UTF8String,
+        sourceLayerID ? std::optional<std::string>(sourceLayerID.UTF8String) : std::nullopt,
+        featureID.UTF8String,
+        mbglState
+    );
+
+    [self setNeedsRerender];
+}
+
+- (nullable NSDictionary<NSString *, id> *)getFeatureState:(id<MLNFeature>)feature sourceID:(NSString *)sourceID {
+    MLNAssertIsMainThread();
+
+    if (!feature || !sourceID) {
+        return nil;
+    }
+
+    NSString *featureID = [NSString stringWithFormat:@"%@", feature.identifier];
```

Comment:
Should we change this to just accept the `feature.identifier` directly? Mapbox seems to just use the identifier: https://docs.mapbox.com/ios/maps/api/11.17.1/documentation/mapboxmaps/mapboxmap/setfeaturestate(sourceid:sourcelayerid:featureid:state:callback:)/

Maybe the caller just stores a list of feature ids instead of the whole thing.